tests/utils/test_file_util.py

```diff
@@ -38,7 +38,16 @@ def test_read_write_file(self):
         content_read = fileutil.read_file(test_file)
         self.assertEquals(content, content_read)
         os.remove(test_file)
-    
+
+    def test_write_empty_file(self):
+        """
+        write_file should ignore contents == None, and do nothing
+        """
+        test_file=os.path.join(self.tmp_dir, self.test_file)
+        fileutil.write_file(test_file, None)
```

Comment:
If the file exists before write_file(path, None) is called, then your new behavior results in the file being untouched. The previous behavior for results in the file being truncated to zero length when append is False. You're changing behavior. If you desire this new behavior, you should have explicit tests that address the permutations.

Reply:
This test is badly named; you're not writing an empty file, you're writing None. `contents=""` is different from `contents=None`.